src/libertem/job/masks.py

```diff
@@ -19,20 +20,24 @@ def _get_masks_for_slice(slice_):
             # .reshape((-1,)) -> like flatten, but no copies
             # should save us one copy as we np.stack() immediately afterwards
             # https://stackoverflow.com/a/28930580/540644
-            slice_.get(mask, signal_only=True).reshape((-1,))
+            slice_.get(mask, signal_only=True).reshape((-1, 1))
```

Comment:
Can you explain why this is necessary? Just a difference between numpy arrays and scipy sparse matrices?

Reply:
@sk1p Sparse matrices can only be reshaped to 2D, not 1D. Technically one could write this somehow that dense arrays get a different treatment here. I felt that it would be cleaner to have equivalent code for sparse and dense.